src/browser/browser.cpp

```diff
@@ -22,13 +22,50 @@
 #include <QQuickView>
 #include <QTimer>
 #include <QUrl>
+#include <QDebug>
 #include <webengine.h>
 #include <webenginesettings.h>
 
+const auto MOZILLA_DATA_DIR = QStringLiteral("/.mozilla/mozembed/");
+const auto MOZILLA_DATA_UA_UPDATE = QStringLiteral("ua-update.json");
+const auto MOZILLA_DATA_UA_UPDATE_SOURCE = QStringLiteral("/usr/share/sailfish-browser/data/ua-update.json.in");
+const auto MOZILLA_DATA_PREFS = QStringLiteral("prefs.js");
+const auto MOZILLA_DATA_PREFS_SOURCE = QStringLiteral("/usr/share/sailfish-browser/data/prefs.js");
+
 BrowserPrivate::BrowserPrivate(QQuickView *view)
     : view(view)
     , closeEventFilter(nullptr)
 {
+    initUserData();
+}
+
+void BrowserPrivate::initUserData()
```

Comment:
This looks good now I think. It does exactly what it should i.e. if .mozembed is gone the data is copied back upon application startup.